### DISCUSSION TITLE
ART for few shot learning

### DISCUSSION THREAD

**Discussion by anukriti15** [Trusted-AI/adversarial-robustness-toolbox#2162] [Q&A]
Can we use the evasion attacks for few-shot learning models? Forward function is the problem as support and query set is involved

**Comment by beat-buesser** [comment]
Hi @anukriti15 My first thought would be that it should be possible. Are you looking for white- or black-box attacks?

**Reply by anukriti15** [reply]
My preference will be white box attacks

**Reply by beat-buesser** [reply]
For white-box attack you will have to create a new estimator specific for your model that internally loads the support set and applies it appropriately in estimator methods like `loss_gradient`, `predict`, etc. For black-box attack you might be able to use the existing estimator `BlackBoxClassifier` and include the support set in the custom function `predict_fn`.

**Reply by anukriti15** [reply]
Thanks a lot. I will attempt this and get back to you for any further queries.

**Reply by anukriti15** [reply]
Also, does ART support adversarial attacks for text or tabular data?

**Reply by beat-buesser** [reply]
There are capabilities for tabular data with continuous features.
